modules/core/src/lib/deck-picker.js

```diff
@@ -26,31 +26,122 @@ import {getClosestObject, getUniqueObjects} from '../utils/query-object';
 import {processPickInfo, getLayerPickingInfo} from '../utils/pick-info';
 
 export default class DeckPicker {
-  constructor(props) {}
+  constructor(props) {
+    this.gl = props.gl;
+    this.layerManager = props.layerManager;
+    this.pickingEvent = null;
+    this.lastPickedInfo = {
+      // For callback tracking and autohighlight
+      index: -1,
+      layerId: null,
+      info: null
+    };
+  }
 
-  // Pick the closest object at the given (x,y) coordinate
-  pickObject(
-    gl,
-    {
+  // Returns a new picking info object by assuming the last picked object is still picked
+  getLastPickedObject({x, y, viewports}) {
```

Comment:
This is more of an additional helper method, I would put this below the pickObject... functions